shell/packages/accounts-email-token/token_server.js

```diff
@@ -146,22 +170,47 @@ var createAndEmailTokenForUser = function (email) {
     userId = Accounts.insertUserDoc(options, user);
   }
 
-  sendTokenEmail(email, token);
+  sendTokenEmail(email, token, linkingIdentity);
 
   return userId;
 };
 
-// method for create user. Requests come from the client.
-// This method will create a user if it doesn't exist, otherwise it will generate a token.
-// It will always send an email to the user
-Meteor.methods({createAndEmailTokenForUser: function (email) {
-  check(email, String);
+Meteor.methods({
+  createAndEmailTokenForUser: function (email, linkingIdentity) {
+    // method for create user. Requests come from the client.
+    // This method will create a user if it doesn't exist, otherwise it will generate a token.
+    // It will always send an email to the user
+
+    check(email, String);
+    check(linkingIdentity, Boolean);
+
+    if (!Accounts.emailToken.isEnabled()) {
+      throw new Meteor.Error(403, "Email identity service is disabled.");
+    }
+    // Create user. result contains id and token.
+    var user = createAndEmailTokenForUser(email, linkingIdentity);
+  },
+
+  linkEmailIdentityToAccount: function (email, token, accountId) {
+    // Links the email identity with address `email` and login token `token` to the account with
+    // ID `accountId`, which must also be the current account.
+    check(email, String);
+    check(token, String);
```

Comment:
Please add a check() for accountId -- even though there's obviously no injection possibility since it is only used in a `!==` comparison, I'd like to as a rule check() everything so that we could, for example, have an automated script verify our checks.

Alternatively, maybe it makes sense to drop accountId as a parameter, since it's not valid to specify anything other than the caller's user ID anyway?